identity/aziot-identityd/src/lib.rs

```diff
@@ -512,8 +514,14 @@ impl UpdateConfig for Api {
     type Error = Error;
 
     async fn update_config(&mut self, new_config: config::Settings) -> Result<(), Self::Error> {
+        // Abort if update fails; i.e. if reconciling identities fails. This may happen if there's
+        // an error in the config or IoT Hub. The user will have to fix the config or IoT Hub and
+        // restart IS.
         self.update_config_inner(new_config, ReprovisionTrigger::ConfigurationFileUpdate)
             .await
+            .expect("failed to reprovision");
```

Comment:
Does this fail even if there is network connectivity failure? The same IS shutdown doesn't happen during normal identity operations. Hence, the question

Reply:
Yes, this would fail if Hub can't be reached to create/delete identities.

Normal identity operations are able to return an error via the HTTP response. Reloading the config can't do that because it's not an API call. It's probably better to abort so that it's obvious something is wrong instead of continuing here.

Reply:
I thought the list of reasons for IS failure was scoped to provisioning, not reconcilation. I would stay consistent with the identity API behavior and not kill the process, if Hub calls fail.

Reply:
Got it. I've updated the behavior so IS doesn't abort on reconciliation failure.